Sources/MapboxCoreNavigation/NavigationRouter.swift

```diff
@@ -0,0 +1,321 @@
+            decoder.userInfo = userInfo
+            
+            guard let disposition = try? decoder.decode(ResponseDisposition.self, from: data) else {
+                let apiError = DirectionsError(code: nil,
+                                               message: nil,
+                                               response: nil,
+                                               underlyingError: result.error as? Error)
+
+                self.complete(requestId: requestId) {
+                    completion(.failure(apiError))
+                }
+                return
+            }
+            
+            guard (disposition.code == nil && disposition.message == nil) || disposition.code == "Ok" else {
+                let apiError = DirectionsError(code: disposition.code,
+                                               message: disposition.message,
+                                               response: nil,
+                                               underlyingError: result.error as? Error)
+
+                self.complete(requestId: requestId) {
+                    completion(.failure(apiError))
+                }
+                return
+            }
+            
+            let result = try decoder.decode(ResponseType.self, from: data)
+            
+            self.complete(requestId: requestId) {
+                completion(.success(result))
+            }
+        } catch {
+            self.complete(requestId: requestId) {
+                let bailError = DirectionsError(code: nil, message: nil, response: nil, underlyingError: error)
+                completion(.failure(bailError))
+            }
+        }
+    }
+    
+    fileprivate func doRequest<ResponseType: Codable>(options: DirectionsOptions,
+                                                      completion: @escaping (Result<ResponseType, DirectionsError>) -> Void) -> RequestId {
+        let directionsUri = settings.directions.url(forCalculating: options).removingSKU().absoluteString
+        var requestId: RequestId!
+        requestsLock.lock()
+        requestId = router.getRouteForDirectionsUri(directionsUri) { [weak self] (result, _) in
+            guard let self = self else { return }
+            
+            self.parseResponse(requestId: requestId,
+                               userInfo: [.options: options,
+                                          .credentials: self.settings.directions.credentials],
+                                          result: result,
+                                          completion: completion)
+        }
+        activeRequests[requestId] = .init(id: requestId,
+                                          router: self)
+        requestsLock.unlock()
+        return requestId
+    }
+}
+
+extension DirectionsProfileIdentifier {
+    var nativeProfile: RoutingProfile {
+        return RoutingProfile(profile: rawValue)
+    }
+}
+
+extension URL {
+    func removingSKU() -> URL {
```

Comment:
Removing the `SKU` token, as NN automatically appends it on their side, which may result in conflict.

Reply:
Do you have an integration test for this? I'm worried that we break encapsulation here by using magic constant "sku". Can we move this logic to Directions library?

Reply:
I don't think it should be a publicly available Directions feature, as it is a part of internal logic communicating w/ NN. Agreed on magic constants usage, but don't know how to avoid it in this case as ideally it should be shared with Directions too.

Reply:
Can it be shared through the environment, the same way we used to share the SKU token with MapboxDirections?

Reply:
I think we could. It would require internal changes on Directions side too, so maybe we could do it as a tailwork?